src/main/java/org/gridsuite/modification/server/NetworkModificationController.java

```diff
@@ -140,296 +96,65 @@ public ResponseEntity<List<UUID>> getModificationGroups() {
-        networkModificationService.updateLineSplitWithVoltageLevelCreation(modificationUuid, lineSplitWithVoltageLevelInfos);
-        return ResponseEntity.ok().build();
-    }
-
-    @PostMapping(value = "/networks/{networkUuid}/line-attach", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
-    @Operation(summary = "attach line to a voltage level")
-    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The line has been attached to voltage level")})
-    public ResponseEntity<List<ModificationInfos>> createLineAttachToVoltageLevelCreation(@Parameter(description = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
-                                                                                          @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId,
-                                                                                          @RequestParam(value = "group", required = false) UUID groupUuid,
-                                                                                          @RequestParam(value = "reportUuid") UUID reportUuid,
-                                                                                          @RequestParam(value = "reporterId") String reporterId,
-                                                                                          @RequestBody LineAttachToVoltageLevelInfos lineAttachToVoltageLevelInfos) {
-        return ResponseEntity.ok().body(networkModificationService.createLineAttachToVoltageLevelCreation(networkUuid, variantId, groupUuid, reportUuid, reporterId, lineAttachToVoltageLevelInfos));
+        @ApiResponse(responseCode = "200", description = "The network modification was created"),
+        @ApiResponse(responseCode = "404", description = "The network or equipment was not found")})
+    public ResponseEntity<List<? extends ModificationInfos>> createNetworkModification(
+            @Parameter(description = "Network ID") @RequestParam("networkUuid") UUID networkUuid,
+            @Parameter(description = "Variant ID") @RequestParam(name = "variantId", required = false) String variantId,
+            @Parameter(description = "Group ID") @RequestParam(name = "groupUuid", required = false) UUID groupUuid,
+            @Parameter(description = "Report ID") @RequestParam("reportUuid") UUID reportUuid,
+            @Parameter(description = "Reporter ID") @RequestParam("reporterId") String reporterId,
+            @RequestBody ModificationInfos modificationInfos) {
+        // temporary switch, should be removed when all kind of modification will use the generic updateModification
+        // PS : same for the wildcard return type (code smell)
+        // PS 2 : the switch can't be in the service because of @Transactional that need to be called from outside the class
```

Comment:
?

Reply:
I think we already talked about, it's for the @Transactional